### PR TITLE
Tenant policy multicast route map policy

### PR DESCRIPTION
**Pull request by akinross**
needed for #426 and https://github.com/CiscoDevNet/ansible-mso/issues/425

### CODE REVIEW

**Comment by akinross** [comment]
requires rebase after https://github.com/CiscoDevNet/ansible-mso/pull/430 is merged

**Review by sajagana — APPROVED** [review]
LGTM!

File: tests/integration/targets/ndo_template/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+
+    - name: Error on not allowed tenant
+      cisco.mso.ndo_template:
+        <<: *create_monitoring_access_template
+        tenant: ansible_test
+        state: present
+      register: err_not_allowed_tenant
+      ignore_errors: true
+
+    - name: Err on template query without template_type
```
Comment (anvitha-jain) [inline tests/integration/targets/ndo_template/tasks/main.yml:490]: Err -> Error

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: tests/integration/targets/ndo_template/tasks/main.yml (hunk 1 above)
Comment (akinross) [reply tests/integration/targets/ndo_template/tasks/main.yml]: Changed and rebased all the PRs that are affected by this

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

File: plugins/modules/ndo_route_map_policy_multicast.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+      source:
+        description:
+        - The data of the entry.
+        type: str
+        required: true
+      rp:
+        description:
+        - The rendezvous point of the entry.
+        type: str
+        aliases: [ rendezvous_point ]
```
Comment (samiib) [inline plugins/modules/ndo_route_map_policy_multicast.py:71]: I think the name and the alias should be switched.
```suggestion
      rendezvous_point:
        description:
        - The rendezvous point of the entry.
        type: str
        aliases: [ rp ]
```

File: plugins/modules/ndo_route_map_policy_multicast.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    - The name of the Multicast Route Map Policy.
+    type: str
+    aliases: [ name ]
+  route_map_policy_uuid:
+    description:
+    - The uuid of the Multicast Route Map Policy.
+    - This parameter is required when the O(route_map_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  description:
```
Comment (samiib) [inline plugins/modules/ndo_route_map_policy_multicast.py:40]: Should we add some aliases for description?

**Review by samiib — CHANGES_REQUESTED** [review]

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 2 above)
Comment (akinross) [reply plugins/modules/ndo_route_map_policy_multicast.py:71]: why?

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 3 above)
Comment (akinross) [reply plugins/modules/ndo_route_map_policy_multicast.py:40]: For multiple other modules ( see template modules where description ) we do not add an alias, so do not see why we should now introduce here. If so perhaps raise an issue and change it consistently everywhere.

**Review by anvitha-jain — APPROVED** [review]
LGTM

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 2 above)
Comment (samiib) [reply plugins/modules/ndo_route_map_policy_multicast.py:71]: As a general option convention, If I was adding the full name as an option/alias, I would use shorter names and alternatives in alias. However, it is up to you as there is no discernible benefit either way.

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 3 above)
Comment (samiib) [reply plugins/modules/ndo_route_map_policy_multicast.py:40]: Its fine to keep it consistent. I was just thinking about what we usually do in `ansible-aci` for description.

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 2 above)
Comment (akinross) [reply plugins/modules/ndo_route_map_policy_multicast.py:71]: ok will keep as is

File: plugins/modules/ndo_route_map_policy_multicast.py (hunk 3 above)
Comment (akinross) [reply plugins/modules/ndo_route_map_policy_multicast.py:40]: Yeah I know there we use `descr`, but I think this is also due to the model in apic using `descr`. Feel free to open an new issue for changing everywhere if you feel this is an improvement.

**Review by samiib — APPROVED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**